searx/version.py

```diff
@@ -1,26 +1,150 @@
-# -*- coding: utf-8 -*-
-'''
-searx is free software: you can redistribute it and/or modify
-it under the terms of the GNU Affero General Public License as published by
-the Free Software Foundation, either version 3 of the License, or
-(at your option) any later version.
-
-searx is distributed in the hope that it will be useful,
-but WITHOUT ANY WARRANTY; without even the implied warranty of
-MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-GNU Affero General Public License for more details.
-
-You should have received a copy of the GNU Affero General Public License
-along with searx. If not, see < http://www.gnu.org/licenses/ >.
-
-(C) 2013- by Adam Tauber, <asciimoo@gmail.com>
-'''
-
-# version of searx
-VERSION_MAJOR = 1
-VERSION_MINOR = 0
-VERSION_BUILD = 0
-
-VERSION_STRING = "{0}.{1}.{2}".format(VERSION_MAJOR,
-                                      VERSION_MINOR,
-                                      VERSION_BUILD)
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# lint: pylint
+# pylint: disable=missing-function-docstring,missing-module-docstring,missing-class-docstring
+
+# some code is from https://github.com/pypa/setuptools_scm/blob/b0b46fd723f6649a7ba4895b005bc89db1529e22/src/setuptools_scm/utils.py  # pylint: disable=line-too-long
+# ( MIT License )
+
+import os
+import shlex
+import subprocess
+
+
+class ShellException(Exception):
+
+    def __init__(self, command, exit_code, stdout, stderr):
+        super().__init__(self, stderr)
+        self.command = command
+        self.exit_code = exit_code
+        self.stdout = stdout
+        self.stderr = stderr
+
+
+def no_git_env(env):
+    # adapted from pre-commit
+    # Too many bugs dealing with environment variables and GIT:
+    # https://github.com/pre-commit/pre-commit/issues/300
+    # In git 2.6.3 (maybe others), git exports GIT_WORK_TREE while running
+    # pre-commit hooks
+    # In git 1.9.1 (maybe others), git exports GIT_DIR and GIT_INDEX_FILE
+    # while running pre-commit hooks in submodules.
+    # GIT_DIR: Causes git clone to clone wrong thing
+    # GIT_INDEX_FILE: Causes 'error invalid object ...' during commit
+    return {
+        k: v
+        for k, v in env.items()
+        if not k.startswith("GIT_")
+        or k in ("GIT_EXEC_PATH", "GIT_SSH", "GIT_SSH_COMMAND")
```

Comment:
Do I miss something, what is the difference of `startswith("GIT_")` and `in ("GIT_EXEC_PATH", "GIT_SSH", "GIT_SSH_COMMAND")`

Reply:
`not k.startswith("GIT_") or k in ...` : 
* it accepts `GIT_EXEC_PATH`, `GIT_SSH`, `GIT_SSH_COMMAND` keys
* and all other keys except the keys that start with `GIT_`.

Reply:
🤦🏻 .. sorry.